compiler/rustc_session/src/options.rs

```diff
@@ -58,10 +58,58 @@ macro_rules! hash_substruct {
     };
 }
 
+macro_rules! gather_tmods {
+    ($_opt_name:ident, $init:expr, $opt_expr:expr, $mods:expr, [SUBSTRUCT], [TMOD]) => {
+        compile_error!("SUBSTRUCT can't be TMOD (target modifier)");
+    };
+    ($opt_name:ident, $init:expr, $opt_expr:expr, $mods:expr, [UNTRACKED], [TMOD]) => {
+        if *$opt_expr != $init {
```

Comment:
Shouldn't we also record a target modifier set to its default value?

Reply:
I am not sure, it just looks strange for me to record "target modifiers" if nothing is modified.

Reply:
I think it's fine to not record the modifier when it's the default, as long as we still catch a mismatch when one crate has the default and the other does not. It would be good to have a test for this.